src/librustdoc/html/render/mod.rs

```diff
@@ -2410,3 +2420,160 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
     }
     out
 }
+
+const MAX_FULL_EXAMPLES: usize = 5;
+
+/// Generates the HTML for example call locations generated via the --scrape-examples flag.
+fn render_call_locations(
+    w: &mut Buffer,
+    cx: &Context<'_>,
+    call_locations: &Option<FnCallLocations>,
+) {
+    let call_locations = match call_locations.as_ref() {
+        Some(call_locations) if call_locations.len() > 0 => call_locations,
+        _ => {
+            return;
+        }
+    };
+
+    // Generate a unique ID so users can link to this section for a given method
+    let id = cx.id_map.borrow_mut().derive("scraped-examples");
+    write!(
+        w,
+        r##"<div class="docblock scraped-example-list">
+          <h1 id="scraped-examples" class="small-section-header">
+             <a href="#{}">Examples found in repository</a>
+          </h1>"##,
+        id
+    );
+
+    // Link to the source file containing a given example
+    let example_url = |call_data: &CallData| -> String {
+        format!(
+            r#"<a href="{root}{url}" target="_blank">{name}</a>"#,
+            root = cx.root_path(),
+            url = call_data.url,
+            name = call_data.display_name
+        )
+    };
+
+    // Generate the HTML for a single example, being the title and code block
+    let write_example = |w: &mut Buffer, (path, call_data): (&PathBuf, &CallData)| {
+        // FIXME(wcrichto): is there a better way to handle an I/O error than a panic?
+        //  When would such an error arise?
+        let contents =
+            fs::read_to_string(&path).expect(&format!("Failed to read file: {}", path.display()));
```

Comment:
I could imagine it being caused by a flaky OS error. Also, if a user deleted the file after `cargo` ran and before `rustdoc`, though that seems pretty rare.

Panicking will cause an ICE bug report message, which is not ideal. I think doing something like this:

https://github.com/rust-lang/rust/blob/9bb77da74dac4768489127d21e32db19b59ada5b/src/librustdoc/html/sources.rs#L123-L130

and then skipping that example would probably be the best solution.

Reply:
Ok I implemented the logic you suggested.